### PR TITLE
Fix reading from an offset in vmemcache_populate_value()

### PR DESCRIPTION
**Pull request by ldorau**
This change is [Reviewable](https://reviewable.io/reviews/pmem/vmemcache/182)

### CODE REVIEW

**Comment by ldorau** [comment]
@marcinslusarz it is a very important bug fix

**Comment by ldorau** [comment]
Rebased

File: src/vmemcache.c [hunk 1]
```diff
@@ -341,36 +341,26 @@ static size_t
...
 
 		if (len > vbufsize)
 			len = vbufsize;
 
-		if (len > left_to_copy)
-			len = left_to_copy;
-
 		if (!no_memcpy)
-			memcpy(vbuf, (char *)ext.ptr + off, len);
+			memcpy(vbuf, (char *)ext.ptr + offset, len);
```
Comment (marcinslusarz) [inline src/vmemcache.c]: What if object spans multiple extents? Don't we need to change the offset in each iteration?

Comment (ldorau) [reply src/vmemcache.c]: It is handled in the lines 351-354.

Comment (marcinslusarz) [reply src/vmemcache.c]: Lines 351-354 handle precedent extents only.

**Comment by ldorau** [comment]
I will add a test for this case

**Comment by ldorau** [comment]
A test added.

**Comment by ldorau** [comment]
@marcinslusarz This the 1st PR to be merged.

**Comment by ldorau** [comment]
@kilobyte please review

**Pull request merged**